### DISCUSSION TITLE
Running parallel tests in Gitlab CI

### DISCUSSION THREAD

**Discussion by clement-hvt** [orchestral/testbench#366] [Q&A]
- Testvench version : 6.27.1
- Laravel: 8/83.27
- PHP Version 7.4.33
- Database Driver: Gitlab CI service mysql:5.7

Hi,
I'd like to run my tests in parallel with the --parallel option in my Gitlab CI but I can't manage it. From the logs I get the impression that the different databases are not created (.*test_1, .*test_2, .*test_3 ...). I'm running my tests on a mysql service offered by gitlab CI (host = `mysql`). Is this possible and does it come from the image I'm using? Or is it not possible to run tests in parallel on a gitlab CI?

This is the command I use: `./vendor/bin/testbench package:test --parallel -c phpunit.xml --log-junit phpunit-report.xml`

Thanks !

**Comment by AJenbo** [comment]
Have you tried using SQLite with `:memory:` rather then MySQL for your tests? This would isolate each job fully.

**Reply by crynobone** [reply]
You cannot run parallel tests using in-memory database as parallel will need to create a separate database for each "worker"

**Reply by AJenbo** [reply]
Maybe "you" can't but I'm doing it just fine and have been doing so for years.
